lib/mutant/parser.rb

```diff
@@ -12,18 +12,18 @@ def initialize
       @cache = {}
     end
 
-    # Root node parsed from file
+    # Parse path into AST
     #
-    # @param [#to_s] path
+    # @param [Pathname] path
     #
     # @return [AST::Node]
     #
     # @api private
-    def parse(path)
+    def call(path)
       @cache.fetch(path) do
-        @cache[path] = Parser::CurrentRuby.parse(File.read(path))
+        @cache[path] = ::Parser::CurrentRuby.parse(path.read)
```

Comment:
Yeah, but the whole AST for this method could be simplified to:

```ruby
def call(path)
  @cache[path] ||= ::Parser::CurrentRuby.parse(path.read)
end
```

I can't think of any metric where that is not better than the current code.